frontend/src/api.js

```diff
@@ -465,4 +467,11 @@ const hasAttachment = response => {
   return dispositionHeader && dispositionHeader.indexOf("attachment") !== -1;
 };
 
+const validateUrl = url => {
```

Comment:
The name `validateUrl` is misleading since it changes the given value. I suggest to change the name to `removeEmptyQueryParams`.
I also suggest to add a function doc comment here to describe that the value is changed instead of returning a copy.
Each regex should have an explanation comment so it's more clear what's going on.